client/src/components/ListPopup.js

```diff
@@ -90,48 +88,38 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const demoProductsArray = [
-  {
-    name: "Levi's 501 crop jean in lightwash",
-    currency: "$",
-    old_price: 9000,
-    price: 5600,
-    url:
-      "https://www.amazon.ca/Levis-Womens-Skinny-Jeans-Matter/dp/B077WTLTM5/ref=sr_1_7?dchild=1&keywords=levis+501+crop+jeans&qid=1603251804&sr=8-7",
-    img_url: levis_501,
-  },
-  {
-    name: "Tommy Jeans '90s Constrast Cropped Hoodie Sweatshirt'",
-    currency: "$",
-    old_price: 12000,
-    price: 9800,
-    url:
-      "https://www.amazon.ca/Tommy-Hilfiger-Hooded-Performance-Jacket/dp/B07BFVPK7K/ref=sr_1_15?dchild=1&keywords=tommy+jeans+hoodie&qid=1603252642&sr=8-15",
-    img_url: tommy_jeans,
-  },
-  {
-    name: "Calvin Klein Performance Women Jacket",
-    currency: "$",
-    old_price: 11000,
-    price: 5600,
-    url:
-      "https://www.amazon.ca/Tommy-Hilfiger-Hooded-Performance-Jacket/dp/B07BFVPK7K/ref=sr_1_15?dchild=1&keywords=tommy+jeans+hoodie&qid=1603252642&sr=8-15",
-    img_url: ck_jacket,
-  },
-  {
-    name: "Levi's 501 crop jean in lightwash",
-    currency: "$",
-    old_price: 9000,
-    price: 5600,
-    url:
-      "https://www.amazon.ca/Levis-Womens-Skinny-Jeans-Matter/dp/B077WTLTM5/ref=sr_1_7?dchild=1&keywords=levis+501+crop+jeans&qid=1603251804&sr=8-7",
-    img_url: levis_501,
-  },
-];
-
-const ListPopup = ({ listTitle, itemCount, listOpen, changeListOpen, addProductOpen, changeAddProductOpen }) => {
+const ListPopup = ({ listId, listTitle, itemCount, listOpen, changeListOpen, addProductOpen, changeAddProductOpen }) => {
+  
   const classes = useStyles();
   const handleClose = () => changeListOpen();
+  const [listToProducts, setListToProducts] = useState([]); // list-to-products table {list_id, product_id}
+  const [products, setProducts] = useState([]); // products table {product_id, img_url, price}
+
+  const getListRelations = async () => {
+    const res = await fetch(`list-to-products/${listId}`)
+    const json = await res.json()
+    setListToProducts(json);
+    console.log(json)
+  }
+
+  const getProducts = async () => {
+    const newProducts = await Promise.all(listToProducts.map(async (relation)=>{      
+      const res = await fetch(`/products/${relation.product_id}`)
+      console.log(res)
+      return res.json(res);
+      
+    }));
+    setProducts(newProducts)
+  }
+
+  useEffect(() => {
+    getListRelations()
+  }, [listId,listOpen]);
+
+  useEffect(() =>{
+    getProducts()
+  },[listToProducts])
```

Comment:
same thoughts here: 
1) we don't want to make API calls every time these change, and
2) rather than using useEffect we want a named function like handleListOpen that gets called at the relevant times.
-but - something like useEffect with listId to update local settings every time the component receives a new listId would make sense